s3/replication/common/experiments/kafka_consumer.py

```diff
@@ -0,0 +1,162 @@
+#
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+# Kafka consumer code
+# Please note : Few changes are required in the replication
+# manager module to add this consumer code.
+# For more details, please refer the feature branch code:
+# https://github.com/priyanka25081999/cortx-multisite/tree/ps/kafka
+
+import json
+from ast import literal_eval
+from s3replicationmanager.prepare_job import PrepareReplicationJob
+# Make sure to install aiokafka module using the command
+# >> pip3 install aiokafka
+# aiokafka version used for testing : aiokafka==0.7.2
+from aiokafka import AIOKafkaConsumer
+from aiokafka.structs import OffsetAndMetadata, TopicPartition
+
+class KafkaMain:
+    consumer = None
+
+    '''
+    Job id to offset mapping
+      {
+        "job_id": { <topic> : <topic_name>,
+                    <partition> : <partition_name>
+                    <offset> : <offset_value>
+                  }
+            ...
+       }
+    '''
+    job_id_to_offset = {}
+    last_commited_offset = 0
+
+    def __init__(self):
+        """Initialize."""
+        # List store topic, partition and offset of each job
+        self._consumer_details = {}
+        self._job_id = 0
+
+    # Initialize the kafka consumer
+    # This method should be called from app.py class on startup.
+    def initialize(self, app):
+        self._app = app
+        KafkaMain.consumer = AIOKafkaConsumer(
+            'FDMI',
+            bootstrap_servers='127.0.0.1:9092',
+            group_id="test",
+            auto_offset_reset='latest',
+            enable_auto_commit=False)
+
+    # Getter methods
+    def get_topic(self, job_id):
+        """Returns the topic."""
+        self._topic = KafkaMain.job_id_to_offset[job_id]["Topic"]
+        return self._topic
```

Comment:
1. Why do we need a class member variable `_topic`? Why don't we directly return the result?
2. When we are accessing the `job_id_to_offset`, why do we need to use "KafkaMain.job_id_to_offset"?
    why not `self.job_id_to_offset[job_id]["Topic"]`?

Reply:
1. Yes, we can directly return the result. Updated the code
2. The job_id_to_offset is a dictionary which maintains the mapping of job id and consumed event's offset with topic and partition details. Yes, we can access and use it as an instance attribute. But, To access it irrespective of object and to maintain the state I am accessing it using KafkaMain.job_id_to_offset.